File: level2/전화번호목록.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool solution(vector<string> phone_book) {
    sort(phone_book.begin(),phone_book.end());
    
    bool answer = true;
    int size = phone_book.size();
    for(int i = 1; i < size; i++){
        int num_len = phone_book[i-1].size();
        
        if(num_len > phone_book[i].size()){
            continue;
        }
        
        bool tmp = true;
        for(int j = 0; j < num_len; j++){
            if(phone_book[i-1][j] != phone_book[i][j]){
                tmp = false;
                break;
            }
        }

        if(tmp){
            answer = false;
            return answer;
        }
    }
    
    return answer;
}

File: level2/전화번호목록.py
def solution(phone_book):
    answer = True
    
    phone_book.sort(key=len)
    
    dic = {phone_book[0]:0}
    for i in phone_book:
        check = 0
        for key in dic:
            if key == i[:len(key)]:
                dic[key] += 1
                check = 1
                if dic[key] > 1:
                    answer = False
                    return answer
        if check == 0:
            dic[i] = 1                
    
    return answer
